CASSANDRA-17009 Adds unit tests to sstableverify operate on SSTables

bhouse99 opened the pull request:
sstableverify tests didn't cover the case of exercising the verifyier for sstables.
Added unit tests to verify this and several corrupt scenarios.

test/data was marked as git ignore, despite it containing files for use.
Added an exception to the gitignore so forcing test data files is no longer necessary.

There are some small bugs found which will be in subsequent jiras.
(Return codes being wrong, etc).

    - Jira : https://issues.apache.org/jira/browse/CASSANDRA-17009

bhouse99 committed:
CASSANDRA-17009 Adds unit tests to sstableverify operate on SSTables

sstableverify tests didn't cover the case of exercising the verifyier for sstables.
Added unit tests to verify this and several corrupt scenarios.

test/data was marked as git ignore, despite it containing files for use.
Added an exception to the gitignore so forcing test data files is no longer necessary.

There are some small bugs found which will be in subsequent jiras.
(Return codes being wrong, etc).

    - Jira : https://issues.apache.org/jira/browse/CASSANDRA-17009

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:
```diff
@@ -0,0 +1,177 @@
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, corruptStatsTable);
+
+        assertEquals(0, tool.getExitCode());
```
Note to self: Should we return 1 here?

bereng replied:
No it's a bug

bhouse99 replied:
fixed

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:
```diff
@@ -0,0 +1,177 @@
+        createAndPopulateTable(keyspaceName, workingTable, x -> {});
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, workingTable);
+        assertEquals(0, tool.getExitCode());
```
`assertOnCleanExit()`?

bhouse99 replied:
fixed

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:
```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * Since we are testing a verifier, we'd like to corrupt files to verify code paths
+     * This function defiintion is used by {@link this#createAndPopulateTable}.
```
defiinition -> definition

bhouse99 replied:
fixed

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:
```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * This function sets up the keyspace, and table schema for a standardCFMD table.
+     * This will also populate the tableiName with a few rows.  After completion the
```
double space before `After`

bhouse99 replied:
fixed

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:
```diff
@@ -0,0 +1,177 @@
+     * @param keyspace name the table should be created in
+     * @param tableName new table name of the standard cfmd table
+     *
+     * @param corruptionFn function called to corrupt or change the contents on disk.  is pased the Cfs of the table name
```
Typos

bhouse99 replied:
fixed

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java:
```diff
@@ -153,4 +153,13 @@ public void testVerboseArg()
+        assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("usage:"));
+        assertThat(tool.getCleanedStderr(), CoreMatchers.containsStringIgnoringCase("Too many arguments"));
+        assertEquals(1, tool.getExitCode());
+    }
```
While we are here I'd look for opportunities to use `assertOnCleanExit()` in this file

bhouse99 replied:
Replaced on all that could have it run

bhouse99 committed:
CASSANDRA-17009 Adds unit tests to sstableverify operate on SSTables mk2

  sstableverify tests didn't cover the case of exercising the verifyier for sstables.
  Added unit tests to verify this and several corrupt scenarios.

  test/data was marked as git ignore, despite it containing files for use.
  Added an exception to the gitignore so forcing test data files is no longer necessary.

        - Jira : https://issues.apache.org/jira/browse/CASSANDRA-17009

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java:
```diff
@@ -0,0 +1,220 @@
+        Assertions.assertThat(tool.getStdout()).contains("Invalid SSTable", corruptDataTable);
+    }
+
+
```
Extra blanks

bereng commented on test/unit/org/apache/cassandra/tools/StandaloneVerifierTest.java:
```diff
@@ -70,6 +70,8 @@ public void testDefaultCall()
         Assertions.assertThat(tool.getCleanedStderr()).isEmpty();
         assertEquals(0, tool.getExitCode());
         assertCorrectEnvPostTest();
+        tool.assertOnCleanExit();
```
This test file doesn't mention the other two and vice-versa
